chain/checkpointing/util.go

```diff
@@ -0,0 +1,288 @@
+package checkpointing
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/Zondax/multi-party-sig/pkg/math/curve"
+	"github.com/btcsuite/btcutil/bech32"
+	"github.com/cronokirby/safenum"
+)
+
+type VerificationShare struct {
+	Share string
+}
+
+type TaprootConfigTOML struct {
+	Thershold          int
+	PrivateShare       string
+	PublicKey          string
+	VerificationShares map[string]VerificationShare
+}
+
+func TaggedHash(tag string, datas ...[]byte) []byte {
```

Comment:
If all these functions are exclusively used in this package, we can probably make them private.